astroid/interpreter/_import/spec.py

```diff
@@ -150,7 +150,15 @@ def find_module(
             if modname in sys.stdlib_module_names or (
                 processed and processed[0] in sys.stdlib_module_names
             ):
-                spec = importlib.util.find_spec(".".join((*processed, modname)))
+                try:
+                    with warnings.catch_warnings():
+                        warnings.filterwarnings("ignore", category=Warning)
+                        spec = importlib.util.find_spec(".".join((*processed, modname)))
+                except ValueError:
```

Comment:
The underlying module is still imported and can raise any exception, e.g. `django.core.exceptions.ImproperlyConfigured`:

```py
  File "/Users/jwalls/django/django/contrib/gis/gdal/libgdal.py", line 62, in <module>
    raise ImproperlyConfigured(
    ...<3 lines>...
    )
django.core.exceptions.ImproperlyConfigured: Could not find the GDAL library (tried "gdal", "GDAL", "gdal3.9.0", "gdal3.8.0", "gdal3.7.0", "gdal3.6.0", "gdal3.5.0", "gdal3.4.0", "gdal3.3.0", "gdal3.2.0", "gdal3.1.0"). Is GDAL installed? If it is, try setting GDAL_LIBRARY_PATH in your settings.
```